modules/dnn/src/layers/einsum_layer.cpp

```diff
@@ -541,7 +541,7 @@ class LayerEinsumImpl CV_FINAL : public EinsumLayer
                 // Use either the preprocessed inputs (if it is available) or the corresponding raw inputs
                 result = pairwiseOperandProcess(!result.empty() ? result : rawInputs[0],
                                                 !result.empty() ? tmpResult : homogenizedInputDims[0],
-                                                (!preProcessedInputs.empty() && !preProcessedInputs[input].empty()) ? preProcessedInputs[input] : rawInputs[input],
+                                                (!preProcessedInputs[input].empty()) ? preProcessedInputs[input] : rawInputs[input],
```

Comment:
Why is pre-check removed?

(anyway it is wrong as it doesn't check vector size vs index)

Reply:
After https://github.com/opencv/opencv/pull/26059 the `preProcessedInputs` is resized, but not reserved. As soon as the layer has inputs, the vector is not empty. I just selected 4.x version.